larwirecell/Components/CookedFrameSource.cxx

```diff
@@ -32,22 +33,76 @@ WireCell::Configuration CookedFrameSource::default_configuration() const
   cfg["tick"] = 0.5 * WireCell::units::us;
   cfg["frame_tags"][0] = "orig"; // the tags to apply to this frame
   cfg["nticks"] = m_nticks;      // if nonzero, truncate or zero-pad frame to this number of ticks.
+
+  // ----- Ewerton: Modify original CookedFrameSource
+  // ----- to read in products for wire-cell imaging
+
+  cfg["wiener_inputTag"] = "";                 // art tag for wiener recob::Wire
+  cfg["gauss_inputTag"] = "";                  // art tag for gauss recob::Wire
```

Comment:
I'll just make one big comment but it applies to essentially this whole diff.

There are two main problems with this PR that must be cleaned up.   They are like we discussed on zoom and I commented on in the PR.

1. This class must work for a wide variety of users and so should not have any application specific information that is hard-wired in C++ or in configuration.  Ie, there should be no configuration keys nor C++ symbols that include the label `wiener` or `bad`, etc.  Instead, the C++ should be purely "data driven" by the configuration.

2. There should be no "modal" `true/false` configuration values.  Instead, if some action is to be taken by the class then the configuration values needed to perform that action will be given.  If no such configuration values are given then the class simply does not take the action.


Specifically:

- Remove all the `*_inputTag` configuration and code that touches them.  There is already a `frame_tags` configuration and that is generic and is interpreted by converting `art::Event` data into frame traces.

- Remove the current `cmm_tag` (and `badmasks_inputTag`) configuration and all code that touches them.  What the user must supply is a instead mapping that associates a CMM **key** (eg `"bad"` to an `art::Event` **label**.  I would call this configuration parameter, perhaps, `cmm_entries`.

- Remove `m_cmm_setup` and instead convert CMM information (using your existing  by iterating on the new `cmm_entries` configuration.  If that configuration is empty then so is the loop body.  The actual conversion code you have at the end of this diff (the loop over `bad_masks`, but again, the symbol "bad" is ... bad) looks okay to me.